instance/tests/models/test_openedx_storage_mixins.py

```diff
@@ -219,3 +280,35 @@ def test_ansible_settings_swift_ephemeral(self):
         instance = OpenEdXInstanceFactory(use_ephemeral_databases=True)
         appserver = make_test_appserver(instance)
         self.check_ansible_settings(appserver, expected=False)
+
+    def test_ansible_settings_s3(self):
+        """
+        Verify Swift Ansible configuration when Swift is enabled.
+        """
+        instance = OpenEdXInstanceFactory(use_ephemeral_databases=False)
+        appserver = make_test_appserver(instance, s3=True)
+        self.check_ansible_settings(appserver, s3=True)
+
+    def test_get_s3_policy(self):
+        """
+        Verify S3 policy is set for correctly
+        """
+        instance = OpenEdXInstanceFactory(use_ephemeral_databases=False)
+        policy = {
+            "Version": "2012-10-17",
+            "Statement": [
+                {
+                    "Effect": "Allow",
+                    "Action": ["s3:ListBucket", "s3:CreateBucket", "s3:DeleteBucket"],
+                    "Resource": ["arn:aws:s3:::{}".format(instance.s3_bucket_name)]
+                },
+                {
+                    "Effect": "Allow",
+                    "Action": [
+                        "s3:*Object*"
+                    ],
+                    "Resource": ["arn:aws:s3:::{}/*".format(instance.s3_bucket_name)]
+                }
+            ]
+        }
+        self.assertEqual(json.dumps(policy, indent=2), instance.get_s3_policy())
```

Comment:
I don't think you need to check for exact strings here.. feels brittle and a bit too much like copy-paste test code.  Maybe just verify that the expected Actions and Resource lines are in place?